lib/sisimai/bite/email/messagelabs.rb

```diff
@@ -0,0 +1,213 @@
+module Sisimai::Bite::Email
+  module MessageLabs
+    # Sisimai::Bite::Email::MessageLabs parses a bounce email which created by
+    # Symantec.cloud: formerly MessageLabs. Methods in the module are called
+    # from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/MessageLabs.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/MAILER-DAEMON[@]messagelabs[.]com/,
+        :subject => %r/\AMail Delivery Failure/,
+      }
+      Re1 = {
+        :begin   => %r|\AContent-Type: message/delivery-status|,
+        :error   => %r/\AReason:[ \t]*(.+)\z/,
+        :rfc822  => %r|\AContent-Type: text/rfc822-headers\z|,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        userunknown: %r/No[ ]such[ ]user/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Symantec.cloud http://www.messagelabs.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+
+      # X-Msg-Ref: server-11.tower-143.messagelabs.com!1419367175!36473369!1
+      # X-Originating-IP: [10.245.230.38]
+      # X-StarScan-Received:
+      # X-StarScan-Version: 6.12.5; banners=-,-,-
+      # X-VirusChecked: Checked
+      def headerlist;  return ['X-Msg-Ref']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Symantec.cloud(MessageLabs)
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-msg-ref']
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        havepassed = ['']
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [102.6/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->